google/cloud/pubsub_v1/publisher/_batch/thread.py

```diff
@@ -226,6 +239,37 @@ def _start_commit_thread(self) -> None:
         )
         commit_thread.start()
 
+    def _start_publish_rpc_span(self) -> None:
+        tracer = trace.get_tracer(self._OPEN_TELEMETRY_TRACER_NAME)
+        links = []
+
+        for wrapper in self._message_wrappers:
+            span = wrapper.create_span
+            # Add links only for sampled spans.
+            if span.get_span_context().trace_flags.sampled:
+                links.append(trace.Link(span.get_span_context()))
+        topic_short_name = self._topic.split("/")[3]
```

Comment:
nit: use the same assert as above here and for project ID?

Reply:
Done.